vspace/src/test/java/edu/asu/diging/vspace/core/services/impl/ModuleManagerTest.java

```diff
@@ -0,0 +1,46 @@
+package edu.asu.diging.vspace.core.services.impl;
+
+import java.util.Optional;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+import edu.asu.diging.vspace.core.data.ModuleRepository;
+import edu.asu.diging.vspace.core.model.IModule;
+import edu.asu.diging.vspace.core.model.impl.Module;
+
+public class ModuleManagerTest {
+
+    @Mock
+    private ModuleRepository mockModuleRepo;
+
+    @InjectMocks
+    private ModuleManager managerToTest = new ModuleManager();
+
+    @Before
+    public void init() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void test_getModule_idExists() {
+        Module newModule = new Module();
+        Optional<Module> mockModule = Optional.of(newModule);
+        Mockito.when(mockModuleRepo.findById(newModule.getId())).thenReturn(mockModule);
+
+        IModule iModuleActual = managerToTest.getModule(newModule.getId());
+        Assert.assertEquals(mockModule.get().getId(), iModuleActual.getId());
+        Assert.assertEquals(mockModule.get().getName(), iModuleActual.getName());
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void test_getModule_idNotExists() throws Exception {
+        Assert.assertEquals(mockModuleRepo.findById(Mockito.anyString()).get(),
+                managerToTest.getModule(Mockito.anyString()).getId());
+    }
```

Comment:
this doesn't make sense. You say that a null pointer exception is expected, but the getModule method does not throw a null pointer. You need to test that the method returns null if the id doesn't exist.